src/cargo/ops/cargo_add/mod.rs

```diff
@@ -98,7 +98,14 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
         .get_table(&dep_table)
         .map(TomlItem::as_table)
         .map_or(true, |table_option| {
-            table_option.map_or(true, |table| is_sorted(table.iter().map(|(name, _)| name)))
+            table_option.map_or(true, |table| {
+                is_sorted(
+                    table
+                        .get_values()
+                        .iter()
+                        .map(|(key, _)| key.first().unwrap().to_string()),
```

Comment:
nit: I feel like `key.remove(0)expect("`get_values` key paths always have at least one key")` would be a bit clearer
- The reason the `unwrap` is safe requires some analysis to re
- Personally, I find `to_owned()` clearer than `to_string()` as `to_string` accepts a larger number of input types to the specific output type.  As an alternative, using `remove` bypasses that completely.

Reply:
Make sense! Thanks!